### DISCUSSION TITLE
Task progress overview in sidebar?

### DISCUSSION THREAD

**Discussion by smartygus** [outline/outline#3674] [Ideas]
I understand that Outline is not a task manager, and I don't really want to use it as one, but the visibility (or lack thereof) of tasks at the moment limits the usability of the simple tasks lists for me.

When viewing a collection the progress (number of tasks completed, total number of tasks, with a small circular gauge) is shown directly as part of the page metadata, which is great:

<img width="936" alt="image" src="https://user-images.githubusercontent.com/706508/174588207-2c721a1d-24bc-4e0a-9481-11b1fa932d61.png">

I would love to have this visibility (whether or not a page has open tasks) directly in the sidebar, and thought that maybe having the small circular gauge next to the title could be a nice addition. I did a couple of basic mockups just so see what it would look like:

<img width="346" alt="Screen Shot 2022-06-20 at 12 56 40" src="https://user-images.githubusercontent.com/706508/174588424-8dd0d718-b012-450f-8696-d36aa8213ca6.png">
<img width="346" alt="Screen Shot 2022-06-20 at 12 55 42" src="https://user-images.githubusercontent.com/706508/174588470-cc7ce5c2-0591-4b13-9d06-feb4acd15138.png">

Any thoughts?

**Comment by tommoor** [comment]
The problem here is more of a technical one, even if we wanted to include this data it unfortunately requires loading the document data to display it.

**Reply by almereyda** [reply]
As per the mock ups as they are presented, it seems to me this, when toggled, would only display for currently active documents.

Is that the initial intention?

**Reply by smartygus** [reply]
> The problem here is more of a technical one, even if we wanted to include this data it unfortunately requires loading the document data to display it.

Oh yeah, you're right. I even read about the parsing in #2320, so naturally the content of each document would need to be loaded to generate the small gauge, I hadn't fully thought of that.

> As per the mock ups as they are presented, it seems to me this, when toggled, would only display for currently active documents.
> Is that the initial intention?

That's a good follow up question @almereyda, thanks for clarifying. Actually no, I just happened to mock it up on the currently-open document. But the real utility (for me at least) of having this info in the sidebar would be to be able to quickly see all documents in a hierarchy that have open tasks.

It would definitely make sense to load these in an async fashion as each level of the hierarchy in a collection becomes visible, but nevertheless it would still require loading the contents of those documents to do that.

Just as a further clarification as well: I'm not talking about using the progress indicator to summarise or add up the tasks of all nested documents of a collection/document – that would require walking the entire tree and loading the contents of each, which would be a not ideal at all – but rather just represent the tasks from each document next to its own title.

@tommoor is the problem as you see it more to do with implementation complexity or more because there's a performance concern from tying a UI element of the sidebar to loading the full contents of each document? (or both?? 😅)
